packages/did/src/DidBatcher/DidBatchBuilder.ts

```diff
@@ -165,22 +168,35 @@ export class DidBatchBuilder {
     }
 
     const nonce = initialNonce || (await this.did.getNextNonce())
+    const batchFunction = atomic
+      ? this.api.tx.utility.batchAll
+      : this.api.tx.utility.batch
+
+    const batchesLength = this.batches.length
+
+    if (!batchesLength) {
+      throw SDKErrors.ERROR_DID_BUILDER_ERROR(
+        'Builder was empty, hence it cannot be consumed.'
+      )
+    }
 
     const signedBatches: Extrinsic[] = await Promise.all(
       this.batches.map(async (batch, index) => {
-        const processedBatch = atomic
-          ? this.api.tx.utility.batchAll(batch.extrinsics)
-          : this.api.tx.utility.batch(batch.extrinsics)
+        // Don't create a new batch if the batch contains only one extrinsic
+        const processedBatch =
+          batch.extrinsics.length > 1
+            ? batchFunction(batch.extrinsics)
+            : (batch.extrinsics.pop() as Extrinsic)
```

Comment:
.pop() modifies the array, so when you use `batch.extrinsics` two lines down, this will not be the same any more (and potentially empty)

Reply:
btw if you simply use array indexing (`batch.extrinsics[0]`) you will also resolve the issue of having to force the type

Reply:
Very good catch! https://github.com/KILTprotocol/sdk-js/pull/473/commits/5ceb1f73e0355bb2bb8fc5a3d8b6c206141a439a